### ISSUE TITLE
Unable to get 2FA code mail, client stops working

### ISSUE THREAD

**Issue by espectalll** [blockchain/My-Wallet-V3#477]
After receiving my first mail confirmation and clicking on its verification link, and immediately after entering the wallet password for the second time, the UI shows up an e-mail code textbox, and each textbox gets disabled. On the (Safari) console, the following error appears:

`Error: undefined is not an object (evaluating 'wallet.my.wallet.bch.getHistory')`

Tested on blockchain.info's My-Wallet-V3 Frontend v1.29.9 and My-Wallet-V3 v3.40.9, running on top of Firefox 57.0.2 and Safari 11.0.2 (13604.4.7.1.3).

## Detailed Firefox log

```
Using My-Wallet-V3 Frontend v1.29.9 and My-Wallet-V3 v3.40.9, connecting to /
landing-554956be23423706c0a01135d41bfa4effea9e13.min.js:367:30061
Server Time offset 57ms - This offset 57
landing-554956be23423706c0a01135d41bfa4effea9e13.min.js:368:3506
Second password required to prepare KV Store
my-wallet-c08cb03fc2c059fc13e85ca992b351b9d584768e.min.js:1:791500
Error: wallet.my.wallet.bch is undefined
Wallet/wallet.fetchAccountInfo/<@https://blockchain.info/wallet/js/wallet-6bf7b6b89ee3871c85ca04f78b3f4f192cf203a8.min.js:11:7787
e/<@https://blockchain.info/wallet/js/landing-554956be23423706c0a01135d41bfa4effea9e13.min.js:133:434
$eval@https://blockchain.info/wallet/js/landing-554956be23423706c0a01135d41bfa4effea9e13.min.js:148:65
$digest@https://blockchain.info/wallet/js/landing-554956be23423706c0a01135d41bfa4effea9e13.min.js:145:123
$evalAsync/<@https://blockchain.info/wallet/js/landing-554956be23423706c0a01135d41bfa4effea9e13.min.js:148:154
f@https://blockchain.info/wallet/js/landing-554956be23423706c0a01135d41bfa4effea9e13.min.js:46:490
mg/k.defer/c<@https://blockchain.info/wallet/js/landing-554956be23423706c0a01135d41bfa4effea9e13.min.js:49:418
landing-554956be23423706c0a01135d41bfa4effea9e13.min.js:120:446
Server Time offset 529.625ms - This offset 1002.25
landing-554956be23423706c0a01135d41bfa4effea9e13.min.js:368:3506
Source map error: SyntaxError: JSON.parse: unexpected character at line 1 column 1 of the JSON data
Resource URL: https://blockchain.info/wallet/js/landing-554956be23423706c0a01135d41bfa4effea9e13.min.js
Source Map URL: angular-sanitize.min.js.map
[Learn More]
```

## Detailed Safari log
```
[Info] Using My-Wallet-V3 Frontend v1.29.9 and My-Wallet-V3 v3.40.9, connecting to / (landing-554956be23423706c0a01135d41bfa4effea9e13.min.js, line 367)
[Log] Server Time offset 26ms - This offset 26 (landing-554956be23423706c0a01135d41bfa4effea9e13.min.js, line 368)
[Warning] Second password required to prepare KV Store (my-wallet-c08cb03fc2c059fc13e85ca992b351b9d584768e.min.js, line 1)
[Error] Error: undefined is not an object (evaluating 'wallet.my.wallet.bch.getHistory')
https://blockchain.info/wallet/js/wallet-6bf7b6b89ee3871c85ca04f78b3f4f192cf203a8.min.js:11:7820
https://blockchain.info/wallet/js/landing-554956be23423706c0a01135d41bfa4effea9e13.min.js:133:435
$digest@https://blockchain.info/wallet/js/landing-554956be23423706c0a01135d41bfa4effea9e13.min.js:145:136
https://blockchain.info/wallet/js/landing-554956be23423706c0a01135d41bfa4effea9e13.min.js:148:163
f@https://blockchain.info/wallet/js/landing-554956be23423706c0a01135d41bfa4effea9e13.min.js:46:497
https://blockchain.info/wallet/js/landing-554956be23423706c0a01135d41bfa4effea9e13.min.js:49:419
	(función anónima) (landing-554956be23423706c0a01135d41bfa4effea9e13.min.js:93)
	(función anónima) (landing-554956be23423706c0a01135d41bfa4effea9e13.min.js:134)
	$digest (landing-554956be23423706c0a01135d41bfa4effea9e13.min.js:145:137)
	(función anónima) (landing-554956be23423706c0a01135d41bfa4effea9e13.min.js:148:164)
	f (landing-554956be23423706c0a01135d41bfa4effea9e13.min.js:46:498)
	(función anónima) (landing-554956be23423706c0a01135d41bfa4effea9e13.min.js:49:420)
[Log] Server Time offset 781.5ms - This offset 1537 (landing-554956be23423706c0a01135d41bfa4effea9e13.min.js, line 368)
```

**Comment by espectalll** [comment]
This issue is likely the same one as https://github.com/blockchain/My-Wallet-V3-Frontend/issues/1409

**Comment by espectalll** [comment]
Issue still exists on My-Wallet-V3 Frontend v1.30.4 (with the server running My-Wallet-V3 v3.41.2). `wallet.my.wallet.bch` is still undefined, logs are about the same. Using a clean web session does not fix the problem.

**Comment by espectalll** [comment]
Tested, it's been fixed on My-Wallet-V3 Frontend v1.30.5 and now I can access from older versions as well.
